src/test/java/sender/BMS_Data_Sender_TestSuite.java

```diff
@@ -0,0 +1,41 @@
+package sender;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class BMS_Data_Sender_TestSuite {

```

Comment:
The most crucial functionality in the sender is its output format. Tests for that are not written here.
In the absence of a test for the output format, any mistakes will only be caught only in integration.